src/SmartTransactionsController.ts

```diff
@@ -670,13 +691,16 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
     } catch (e) {
       console.error('provider error', e);
     }
-    const nonceLock = await this.getNonceLock(txParams?.from);
+    const requiresNonce = !txParams.nonce;
+    const nonceLock = requiresNonce
+      ? await this.getNonceLock(txParams?.from)
+      : undefined;
     try {
-      const nonce = hexlify(nonceLock.nextNonce);
+      const nonce = nonceLock ? hexlify(nonceLock.nextNonce) : txParams.nonce;
       if (txParams && !txParams?.nonce) {
         txParams.nonce = nonce;
       }
-      const { nonceDetails } = nonceLock;
+      const nonceDetails = nonceLock ? nonceLock.nonceDetails : {};
```

Comment:
This seems like can be refactored in a simpler if case:

```ts
    const requiresNonce = !txParams.nonce;
    let nonce;
    let nonceLock;
    let nonceDetails = {};

    
    if (requiresNonce) {
      nonceLock = await this.getNonceLock(txParams?.from)
      nonce = hexlify(nonceLock.nextNonce)
      nonceDetails = nonceLock.nonceDetails
      if (txParams) txParams.nonce ??= nonce;
    }
  ```
  
  I'm not familiar with the returned types though

Reply:
Will simplify it.